pkg/volume/volume.go

```diff
@@ -28,6 +28,28 @@ import (
 type Volume interface {
 	// GetPath returns the directory path the volume is mounted to.
 	GetPath() string
+
+	// VolumeMetricsProvider embeds methods for exposing Volume metrics (e.g. capacity).
+	VolumeMetricsProvider
+}
+
+// VolumeMetricsProvider exposes metrics (e.g. capacity) related to a Volume.
+type VolumeMetricsProvider interface {
+	// GetCapacityMetrics returns the CapacityMetrics for this Volume.  Maybe expensive for some implementations.
+	GetCapacityMetrics() (*CapacityMetrics, error)
+}
+
+// CapacityMetrics represents available and consumed capacity for the Volume.
+type CapacityMetrics struct {
+	// VolumeBytesUsed represents the total bytes used under the directory path that the Volume is mounted.
+	// Note: For block devices this maybe more than the total size of the files.  See `man du` for details.
+	VolumeBytesUsed uint64
+
+	// FileSystemBytesUsed represents the total bytes used on the filesystem that the Volume path resides.
```

Comment:
Renamed fields per Vish's suggestion and used a lot of his language in the comments.  PTAL.